### ISSUE TITLE
Multiple calls to `.then()` are not run properly (#v2)

### ISSUE THREAD

**Issue by psaxton** [vlucas/frisby#352]
When there are multiple calls to `.then()` in a frisby test, inner functions are not run properly. This can be demonstrated in the existing test contained in spec/frisby_spec.js:101-123:

```
  it('should use new responseBody when returning another Frisby spec inside then() with multiple specs chained', function (doneFn) {
    mocks.use(['getUser1', 'getUser2WithDelay']);

    frisby.get(testHost + '/users/1')
      .expect('jsonContains', { id: 1 })
      .then(function () {
        mocks.use(['getUser1WithDelay']);

        return frisby.get(testHost + '/users/1')
          .expect('jsonContains', { id: 2 });                 // __Line 110__: This should cause the test to fail
      })
      .then(function (user1) {
        expect(user1.id).toBe(1);
      })
      .then(function () {
        return frisby.get(testHost + '/users/2')
          .expect('jsonContains', { id: 2 });
      })
      .then(function (user2) {
        expect(user2.id).toBe(2);
      })
      .done(doneFn);
  });
```

Line 110 has a typo in which the actual `id` should be 1, however the test still passes. Changing the `id` to 1 does not affect the outcome of the test.

**Comment by psaxton** [comment]
I have fixed this issue in pull request #350. This issue should be closed once the pull request has been reviewed and accepted.

**Comment by psaxton** [comment]
Appears fixed in b33513d. Closing issue.
